### DISCUSSION TITLE
What is the correct way to initialise Supabase JS client in web worker?

### DISCUSSION THREAD

**Discussion by jbasko** [supabase/supabase#12355] [Questions] [answered]
How do you share the Supabase client session created (acquired) in the main thread with the worker thread? I am trying by sending a message to the worker. The worker is then able to call `supabase.auth.setSession()` and it seems okay. However, when I  try `getSession` then I get back nulls.

The things I send across to worker are `url`, `key` (I tried both the `anon` key and the changed access token)`, `access_token` and `refresh_token`. Then in the worker, I do:

```javascript
worker.supabase = createClient(url, key, {
    detectSessionInUrl: false,
})
const {data, error} = await worker.supabase.auth.setSession({access_token, refresh_token})
const getSessionResult = await worker.supabase.auth.getSession()
```

`getSessionResult` equals to `{data: {session: null}, error: null}`.

Can one have two clients in the same origin? Could it be that this one conflicts somehow with the one in the main thread? In future I will have just one, but now I want to gradually move to web worker.

**Comment by GaryAustin1** [comment, marked as answer]
You need to also have persistSession:false in the createClient most likely.   If you don't have localStorage currently setSession/getSession will null out the session.
https://github.com/supabase/gotrue-js/issues/539

**Reply by jbasko** [reply]
Thank you @GaryAustin1 ! I needed `persistSession: false`, but after failing immediately by doing that and seeing how it has helped others in the thread I double checked the spelling and realised I missed the `auth` wrapping of options:

```
worker.supabase = createClient(url, key, {
    auth: {
        detectSessionInUrl: false,
        persistSession: false,
    },
})
```
